src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_validators.py

```diff
@@ -3,8 +3,110 @@
 # Licensed under the MIT License. See License.txt in the project root for license information.
 #---------------------------------------------------------------------------------------------
 
+import math
 import random
+import time
 
-def nsg_name_validator(namespace):
+from azure.cli.core.commands.arm import resource_id
+
+def validate_nsg_name(namespace):
     namespace.network_security_group_name = namespace.network_security_group_name \
         or '{}NSG{}'.format(namespace.vm_name, random.randint(1, 999))
+
+def validate_vm_nics(namespace):
+    from azure.cli.core.commands.client_factory import get_subscription_id
+    nics_value = namespace.network_interface_ids
+    nics = []
+
+    if not nics_value:
+        namespace.network_interface_type = 'new'
+        return
+
+    namespace.network_interface_type = 'existing'
+
+    if not isinstance(nics_value, list):
+        nics_value = [nics_value]
+
+    for n in nics_value:
+        nics.append({
+            'id': n if '/' in n else resource_id(name=n,
+                                                 resource_group=namespace.resource_group_name,
+                                                 namespace='Microsoft.Network',
+                                                 type='networkInterfaces',
+                                                 subscription=get_subscription_id()),
+            'properties': {
+                'primary': nics_value[0] == n
+            }
+        })
+
+    namespace.network_interface_ids = nics
+    namespace.network_interface_type = 'existing'
+
+    namespace.public_ip_address_type = 'none'
+
+def validate_default_vnet(namespace):
+    ns = namespace
+    if not ns.virtual_network and not ns.virtual_network_type:
+        from azure.mgmt.network import NetworkManagementClient
+        from azure.mgmt.resource.resources import ResourceManagementClient
+        from azure.cli.core.commands.client_factory import get_mgmt_service_client
+
+        resource_client = get_mgmt_service_client(ResourceManagementClient)
+        vnet_client = get_mgmt_service_client(NetworkManagementClient).virtual_networks
+
+        rg = resource_client.resource_groups.get(ns.resource_group_name)
+        location = ns.location or rg.location # pylint: disable=no-member
+
+        # find VNET in target resource group that matches the VM's location
+        vnet = next((v for v in vnet_client.list(rg.name) if v.location == location), None) # pylint: disable=no-member
+
+        # TODO: if none found, find qualifying VNET in subscription (like portal)
```

Comment:
For now I removed it. The user can specify the subnet ID if they desire. Grabbing a VNET out of a subscription just seems dangerous from a CLI perspective. If we want to change it later we can.